alerting/src/main/kotlin/org/opensearch/alerting/InputService.kt

```diff
@@ -209,7 +207,9 @@ class InputService(
                 // Monitor runner will send transport request to check permission first. If security plugin response
                 // is yes, user has permission to query AD result. If AD role filter enabled, we will add user role
                 // filter to protect data at user role level; otherwise, user can query any AD result.
-                addUserBackendRolesFilter(monitor.user, searchRequest.source())
+                if (getADBackendRoleFilterEnabled(clusterService, settings)) {
```

Comment:
Can `getADBackendRoleFilterEnabled` should be made more generic like `getClientPluginRoleFilter` so that we can supply the AD filter path just for that one use but the method itself can be used for other setting check